File: leetcode-submissions/1128-remove-all-adjacent-duplicates-in-string/solution.py
class Solution(object):
    def removeDuplicates(self, s):
        """
        :type s: str
        :rtype: str
        """


        n = len(s)
        if n <= 1:
            return s
        stack = [s[0]]
        

        for i in range(1, n):
            if stack and s[i] == stack[-1]:
                stack.pop()
            else:
                stack.append(s[i])
        
        return "".join(char for char in stack)
        


        
        

File: leetcode-submissions/1128-remove-all-adjacent-duplicates-in-string/solution.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> stack = new Stack<>();
        char[] chars = s.toCharArray();
        stack.push(chars[0]);
        for(int i = 1; i < chars.length; i++) {
            if (!stack.isEmpty() && stack.peek() == chars[i]) {
                stack.pop();
            } else {
                stack.push(chars[i]);
            }
        }

        String ans = "";
        while(!stack.isEmpty()) {
            ans = stack.pop() + ans;
        }
        return ans;    
    }
}
